frontend/test/metabase/scenarios/dashboard/actions-on-dashboards.cy.spec.js

```diff
@@ -217,6 +220,247 @@ const MODEL_NAME = "Test Action Model";
+          cy.findByPlaceholderText("Datetimetz").type("2020-03-01T12:00:00");
+          cy.findByPlaceholderText("Time").type("12:57:57");
+          cy.findByPlaceholderText("Timestamp").type("2020-03-01T12:00:00");
+          cy.findByPlaceholderText("Timestamptz").type("2020-03-01T12:00:00");
+
+          cy.button("Save").click();
+        });
+
+        cy.wait("@executeAPI");
+
+        queryWritableDB(
+          `SELECT * FROM ${TEST_COLUMNS_TABLE} WHERE string = 'Zany Zebras'`,
+          dialect,
+        ).then(result => {
+          expect(result.rows.length).to.equal(1);
+          const row = result.rows[0];
+
+          expect(row.uuid).to.equal("a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a15");
+
+          expect(row.integer).to.equal(-20);
+          expect(row.integerUnsigned).to.equal(20);
+          expect(row.tinyint).to.equal(101);
+          expect(row.tinyint1).to.equal(1);
+          expect(row.smallint).to.equal(32767);
+          expect(row.mediumint).to.equal(8388607);
+          expect(row.bigint).to.equal(
+            dialect === "mysql" ? 922337204775 : String(922337204775), // the pg driver makes this a string
+          );
+          expect(row.float).to.equal(3.4);
+          expect(row.double).to.equal(1.79769313486);
+          expect(row.decimal).to.equal("123901.21"); // js needs this to be a string
+
+          expect(row.boolean).to.equal(dialect === "mysql" ? 1 : true);
+
+          expect(row.string).to.equal("Zany Zebras");
+          expect(row.text).to.equal("Zany Zebras");
+
+          expect(row.date).to.include("2020-02-01"); // js converts this to a full date
+
+          // timezones are problematic here
+          expect(row.datetime).to.include("2020-03-01");
+          expect(row.datetimeTZ).to.include("2020-03-01");
+          expect(row.time).to.include("57:57");
+          expect(row.timestamp).to.include("2020-03-01");
+          expect(row.timestampTZ).to.include("2020-03-01");
+        });
+      });
+
+      it("does not show json, enum, or binary columns for implicit actions", () => {
+        createModelFromTable(TEST_COLUMNS_TABLE);
+        cy.get("@modelId").then(id => {
+          createImplicitAction({
+            kind: "create",
+            model_id: id,
+          });
+        });
+
+        createDashboardWithActionButton({
+          actionName: "Create",
+          idFilter: true,
+        });
+
+        clickHelper("Create");
+
+        modal().within(() => {
+          cy.findByPlaceholderText("Uuid").should("be.visible");
+          cy.findByPlaceholderText("Json").should("not.exist");
+          cy.findByPlaceholderText("Jsonb").should("not.exist");
+          cy.findByPlaceholderText("Binary").should("not.exist");
+
+          if (dialect === "mysql") {
+            // we only have enums in postgres as of Feb 2023
+            cy.findByPlaceholderText("Enum").should("not.exist");
+          }
+        });
```

Comment:
I'm worried these tests might not scale well when we add new driver support; it can become difficult to follow conditional blocks per each dialect. I don't think it's a blocker right now, but I wonder if we can do something now to prevent this?

Reply:
Open to ideas, but I think the ideal is we get all the drivers to parity. This was kind of an 11th hour discovery that enum support is different, and I wanted to make sure it's captured somewhere.

Reply:
Yeah, achieving identical behavior across drivers would be ideal, for sure. So far, I can't think of anything better than splitting PSQL and MySQL into different test suites, but it comes with its own downsides (potentially a bigger pain to maintain, and less clear differences between drivers). Unless anyone has better ideas, I'm fine keeping it as is until we actually hit the problem